clippy_lints/src/methods/unnecessary_filter_map.rs

```diff
@@ -36,9 +36,20 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'tcx>, a
                 ControlFlow::Continue(Descend::Yes)
             }
         });
-
         let in_ty = cx.typeck_results().node_type(body.params[0].hir_id);
         let sugg = if !found_filtering {
+            // Check if the closure is .filter_map(|x| Some(x))
+            if name == "filter_map"
+                && let hir::ExprKind::Call(expr, args) = body.value.kind
+            {
+                if is_res_lang_ctor(cx, path_res(cx, expr), OptionSome)
+                    && arg_id.ty_def_id() == args[0].hir_id().ty_def_id()
+                {
+                    if let hir::ExprKind::Path(_) = args[0].kind {
+                        span_lint(cx, UNNECESSARY_FILTER_MAP, arg.span, &format!("{name} is unnecessary"))
```

Comment:
it just occurs to me that... could this be `span_lint_with_sugg` instead :thinking: , suggesting removing the `filter_map`? 

same thing applies [here](https://github.com/rust-lang/rust-clippy/pull/12766/files#diff-e560e6f117247b6eaa96b371723c123dcefb6eefcf88864d93698a854a0f513aL55-L64), while suggest replacing the method call, with `MaybeIncorrect` applicability for both ofc...